src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,78 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+	public delegate void RawGamepadCallback (RawGamepad raw_gamepad);
+
+	private static GamepadMonitor instance;
+
+	private GenericSet<Gamepad?> gamepads;
+	private GenericSet<RawGamepad?> unmapped_raw_gamepads;
+
+	private GamepadMonitor () {
+		if (gamepads == null)
```

Comment:
No need to check: it is guaranteed to be `null` as it is an instance field and we are in the constructor.